components/Typography/src/UIFont+MaterialScalable.m

```diff
@@ -0,0 +1,121 @@
+
+- (UIFont *)mdc_scaledFontForSizeCategory:(UIContentSizeCategory)sizeCategory {
+  if (!self.mdc_scalingCurve) {
+    return self;
+  }
+
+  NSNumber *fontSizeNumber;
+  if (sizeCategory) {
+    fontSizeNumber = self.mdc_scalingCurve[sizeCategory];
+  }
+
+  // If you have queried the table for a sizeCategory that doesn't exist, we will return the
+  // traits for XXXL.  This handles the case where the values are requested for one of the
+  // accessibility size categories beyond XXXL such as
+  // UIContentSizeCategoryAccessibilityExtraLarge.  Accessbility size categories are only
+  // defined for the Body Font Style.
+  if (fontSizeNumber == nil) {
+    fontSizeNumber = self.mdc_scalingCurve[UIContentSizeCategoryExtraExtraExtraLarge];
+  }
+
+  // Guard against broken / incomplete scaling curves by returning self if fontSizeNumber is nil.
+  if (fontSizeNumber == nil) {
+    return self;
+  }
+
+  CGFloat fontSize = (CGFloat)fontSizeNumber.doubleValue;
+  UIFont *scaledFont = [UIFont fontWithDescriptor:self.fontDescriptor size:fontSize];
+  scaledFont.mdc_scalingCurve = self.mdc_scalingCurve;
+
+  return scaledFont;
+}
+
+- (UIFont *)mdc_scaledFontForCurrentSizeCategory {
+  // If we are within an application, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    sizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  return [self mdc_scaledFontForSizeCategory:sizeCategory];
+}
+
+- (nonnull UIFont *)mdc_scaledFontAtDefaultSize {
+  return [self mdc_scaledFontForSizeCategory:UIContentSizeCategoryLarge];
+}
+
+- (CGFloat)scaledValueForValue:(CGFloat)value {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory defaultSizeCategory = UIContentSizeCategoryLarge;
+  UIContentSizeCategory currentSizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    currentSizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    currentSizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  NSNumber *defaultFontSizeNumber = self.mdc_scalingCurve[defaultSizeCategory];
+
+  NSNumber *currentFontSizeNumber = self.mdc_scalingCurve[currentSizeCategory];
+  // If you have queried the table for a sizeCategory that doesn't exist, we will return the
+  // traits for XXXL.  This handles the case where the values are requested for one of the
+  // accessibility size categories beyond XXXL such as
+  // UIContentSizeCategoryAccessibilityExtraLarge.
+  if (currentFontSizeNumber == nil) {
+    currentFontSizeNumber = self.mdc_scalingCurve[UIContentSizeCategoryExtraExtraExtraLarge];
+  }
```

Comment:
Similarly to the above comment; can we delete this fallback behavior?

Reply:
Don.e